CDCgov/cdc-open-viz
# enhancement: [Dev-8008] trendline bar data driven switch

In storybook under Chart, there are "Line Chart Two Points New Chart" and "Line Chart Two Points Regression Test"
 
Both have the same dataset but the "Regression Test" does not have the config property of `allowLineToBarGraph` This is to simulate how current Line Charts will respond to the update. You will see that the line chart will persist even when there are only 2 data points. You will have to go into the editor, under the General Panel and select the "Convert to Bar Graph" checkbox for the conversion to take place.   

Testing: 
1. Select the "Line Chart Two Points New Chart" under Chart. 
2. Switching the filter from Home to Work and back again should show the Line turning to a Bar chart when there are 2 points and back to a line when there are more than 2 points. 
3. Under Storybook controls, turn isEditor to true
4. In the General Panel the Chart type should be set to Line
5. By toggling the "CONVERT TO BAR GRAPH" checkbox, the graph with 2 points should toggle between a line and bar graph.
6. All other functionality remains the same.
7. Switch Chart Type to Bar
8. Test changing the Subtype, orientation, bar style, and any other Bar properties. 
9. Then change Chart Type back to Line and all of the Bar properties should be reset
10. The Bar Chart that is shown when Line Chart is selected should be a Subtype of Regular, Orientation of Vertical, and the Bar Style of Flat.

The previous steps can be repeated on the "Line Chart Two Points Regression Test." The only difference will be, on initial load there will be no bar charts with the 2 data points. Once the isEditor is set to true and the "CONVERT TO BAR GRAPH" checkbox is checked, then everything should be identical to the "Line Chart Two Points New Chart" steps. 

I also made a change in LinearChart that will be systemic for all charts. `if (orientation === 'vertical' && max - min < 3) return formatNumber(tick, 'left', shouldAbbreviate, false, false, '1')`

The above change will make it so if the difference of max and min value of the yAxis is less than 3, then the yAxis points will round to the .5 . I noticed in the example that the max yAxis was 12 and the min was 10. The left axis had 5 ticks and it showed as [10, 10, 11, 11, 12]. After that change it now show as [10, 10.5, 11, 11.5, 12]
